Open.h

```diff
@@ -3,16 +3,206 @@
 #include "HashTable.h"
 #include "Product.h"
 #include <iostream>
-#include <map>
+#include <vector>
+#include <cmath>
+#include <string>
 
 template<typename Key, typename Value>
 class OpenMap: public HashTable<Key, Value> {
-    public:
-        OpenMap() = default;
+private:
+    double MAX_LOAD_FACTOR = 0.7;
 
-        Value get(Key k) = 0;
+    struct KeyValuePair{
+        KeyValuePair(Key k, Value v) {
+            key = k;
+            value = v;
+            valid = true;
+        }
+        KeyValuePair(){
+            key = Key{};
+            value = Value{};
+            valid = false;
+        }
+        Key key;
+        Value value;
+        bool valid;
+    };
 
-        void put(Key k, Value v) = 0;
+    //this is the operation done on any index that collides
+    uint64_t iterate(uint64_t index){
+        return index*index;
```

Comment:
This is not guaranteed to visit every location.  We talked about index + hashmap.size()-1.  Or if the size is always a power of 2, choose a reasonably large prime to space out consecutive probes.